app/frontend/src/features/profile/hooks/useLanguages.ts

```diff
@@ -0,0 +1,28 @@
+import { useQuery } from "react-query";
+import { service } from "service";
+
+export const useLanguages = () => {
+  const { data: { languages, languagesLookup } = {}, ...rest } = useQuery(
+    "getLanguages",
+    () =>
+      service.resources.getLanguages().then((result) =>
+        result.languagesList.reduce(
+          (languagesResult, { code, name }) => {
+            languagesResult.languages[code] = name;
+            languagesResult.languagesLookup[name] = code;
+            return languagesResult;
+          },
+          {
+            languages: {} as { [code: string]: string },
+            languagesLookup: {} as { [name: string]: string },
+          }
```

Comment:
Instead of doing this re-mapping/data wrangling on the frontend, couldn't the backend just return this bidirectional map in the right format? This seems like a task more suited to the backend to me (literally two hardcoded resource files would do the trick?)...

Reply:
@aapeliv, @kalvdans what do you think about that?

Reply:
Protobuf has no native support for dicts, so I'm afraid all lookup tables has to be constructed from lists in the frontend.

Reply:
Protobuf does actually have a `map` type that is an associative array, we should be using that. It's actually quite literally the same wire encoding but with some helper functions. Let me see.

Reply:
I did it using maps here: #1330, what do you think?